### DISCUSSION TITLE
Is it possible to get unsent HTTP/2 requests?

### DISCUSSION THREAD

**Discussion by OldMansKid** [hyperium/hyper#3561] [Q&A] [answered]
Hello again, I have the intention to implement retry HTTP/2 requests after receiving a GOAWAY frame in my HTTP/2 interception proxy. I checked H2 documentation but it seems to me that neither SendRequest nor SendStream have an API to return the request passed to them. I wonder if it's currently possible to implement retry without cloning every request, that's something I don't want to do.

**Comment by seanmonstar** [comment, marked as answer]
It is not currently possible. You may clone the requests. You can check out the the `tower::retry` middleware that does a lot of that for you, if you define a `Policy` for what requests/responses/errors can be retried.

**Reply by OldMansKid** [reply]
Thanks for the info. I prefer more control, the alternative is to forward GOAWAY to clients, usually browsers. But I need to research if it's worth the effort to switch to H2.
